### ISSUE TITLE
Specific Requirements for losses and metrics

### ISSUE THREAD

**Issue by AakashKumarNain** [poets-ai/elegy#54]
Is there any template that we should refer to while creating a PR for a new loss function or a new metric? For example, does the loss has to be inherited from some base class, etc, etc.?

**Comment by cgarciae** [comment]
Hey @AakashKumarNain, I think we need to create a Contributor Guide with some basic templates and guidelines.

If you want to create a Metric or Loss you have to inherit from either `elegy.metrics.metric.Metric` or `elegy.losses.loss.Loss`. I would suggest you use an existing metric as a template and if that metric is in Keras try to port the signature + documentation.

Feel free to send a Hey @AakashKumarNain, I think we need to create a Contributor Guide with some basic templates and guidelines.

If you want to create a Metric or Loss you have to inherit from either `elegy.metrics.metric.Metric` or `elegy.losses.loss.Loss`. I would suggest you use an existing metric as a template and if that metric is in Keras try to port the signature + documentation.

Feel free to send a WIP PR and we can help you out sort the details.

**Comment by AakashKumarNain** [comment]
Perfect. Thanks for the information. I will send a WIP PR then

**Comment by cgarciae** [comment]
Checkout the new [Contributing Guide](https://poets-ai.github.io/elegy/guides/contributing/)
